tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/PackagingInitTestCase.java

```diff
@@ -0,0 +1,379 @@
+        // Test ballerina init
+        getNewInstanceOfBallerinaServer();
+        Path projectPath = tempProjectDirectory.resolve("firstTestWithPackagesMain");
+
+        String[] clientArgsForInit = {"-i"};
+        String[] options = {"\n", "\n", "\n", "m\n", "newpkg\n", "f\n"};
+        ballerinaServer.runMainWithClientOptions(clientArgsForInit, options, getEnvVariables(), "init",
+                                                 projectPath.toString());
+
+        Assert.assertTrue(Files.exists(projectPath.resolve("newpkg").resolve("main.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("newpkg").resolve("tests").resolve("main_test.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("main.bal")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("foo").resolve("tests").resolve("main_test.bal")));
+
+        // Test ballerina build
+        getNewInstanceOfBallerinaServer();
+        ballerinaServer.runMain(new String[0], getEnvVariables(), "build", projectPath.toString());
+        Assert.assertTrue(Files.exists(projectPath.resolve("target").resolve("newpkg.balx")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("target").resolve("foo.balx")));
+        Assert.assertTrue(Files.exists(projectPath.resolve(".ballerina").resolve("repo").resolve(getOrgName())
+                                                  .resolve("foo").resolve("0.0.1").resolve("foo.zip")));
+        Assert.assertTrue(Files.exists(projectPath.resolve(".ballerina").resolve("repo").resolve(getOrgName())
+                                                  .resolve("newpkg").resolve("0.0.1").resolve("newpkg.zip")));
+
+
+        // Test ballerina run on the new package
+        runMainFunction(projectPath, "newpkg");
+
+        // Test ballerina run with balx
+        runMainFunction(projectPath, projectPath.resolve("target").resolve("newpkg.balx").toString());
+    }
+
+    @Test(description = "Test creating a project with invalid options")
+    public void testInitWithInvalidOptions() throws Exception {
+        // Test ballerina init
+        getNewInstanceOfBallerinaServer();
+        Path projectPath = tempProjectDirectory.resolve("testsWithoutPackage");
+        Files.createDirectories(projectPath);
+
+        String[] clientArgsForInit = {"-i"};
+        String[] options = {"\n", "\n", "\n", "123\n", "jkl\n", "f\n"};
+        ballerinaServer.runMainWithClientOptions(clientArgsForInit, options, getEnvVariables(), "init",
+                                                 projectPath.toString());
+
+        Assert.assertTrue(Files.exists(projectPath.resolve(".ballerina")));
+        Assert.assertTrue(Files.exists(projectPath.resolve("Ballerina.toml")));
+    }
+
+    /**
+     * Get new instance of the ballerina server.
+     *
+     * @throws BallerinaTestException
+     */
+    private void getNewInstanceOfBallerinaServer() throws BallerinaTestException {
```

Comment:
Lets rename this to createNewBallerinaServer ? Does it make sense ?

Reply:
Renamed to createNewBallerinaServer with 9da49db